q2_moshpit/eggnog/_utils.py

```diff
@@ -26,3 +44,140 @@ def _parse_build_diamond_db_params(arg_key, arg_val) -> List[str]:
+
+    # prepare an HMM database for faster hmmscan searches
+    print(colorify("Preparing HMM database..."))
+    run_command(cmd=["hmmpress", hmms_merged_p])
+    shutil.move(hmms_merged_p, f"{str(hmm_db_obj)}/{taxon_id}.hmm")
+
+    return idmap_obj, hmm_db_obj, pressed_hmm_db_obj
+
+
+def _download_fastas_into_hmmer_db(taxon_id: int):
+    fastas_obj = ProteinsDirectoryFormat()
+    with tempfile.TemporaryDirectory() as tmp:
+        try:
+            run_command(cmd=[
+                "wget", "-O", f"{tmp}/{taxon_id}_raw_algs.tar",
+                f"{COMMON_URL}/{taxon_id}/{taxon_id}_raw_algs.tar",
+            ])
+        except subprocess.CalledProcessError as e:
+            raise Exception(
+                "Error downloading seed-alignments. "
+                f"wget error code: {e.returncode}"
+            )
+
+        # Extracting
+        print(colorify("Decompressing..."))
+        run_command(
+            cmd=["tar", "xf", f"{taxon_id}_raw_algs.tar"],
+            cwd=tmp
+        )
+
+        files = glob.glob(f"{tmp}/{taxon_id}/*.gz")
+
+        # Extract, remove '-' and save to hmmer_db location
+        print(colorify("Processing FASTA files (this can take a while)... "))
+        for fpi in tqdm(files):
+            new_name = os.path.basename(fpi).replace(".raw_alg.faa.gz", ".fa")
+            fpo = os.path.join(str(fastas_obj), new_name)
+            with gzip.open(fpi, "rt") as f_in, open(fpo, "w") as f_out:
+                content = f_in.read()
+                content = content.replace("-", "")
+                f_out.write(content)
+
+    return fastas_obj
+
+
+def _merge_hmms_and_write_idmap(hmms_merged_p, idmap_p, taxon_id, tmp):
+    # Open output files
+    with open(hmms_merged_p, "a") as hmms, open(idmap_p, "a") as idmap:
+
+        # Iterate through all decompressed files
+        for root, dirnames, files in os.walk(f"{tmp}/{taxon_id}"):
+            for i, file in tqdm(enumerate(files, start=1), total=len(files)):
+                if file.endswith(".hmm"):  # process hmm files
+                    with open(f"{root}/{file}", "r") as hmm_file:
+                        for line in hmm_file:
+                            if line.startswith("NAME "):
+                                line = re.sub(
+                                    r"\.faa\.final_tree(\.fa)?", "", line
+                                )
+                                id = line.replace("NAME  ", "", 1)
+                                idmap.write(f"{i} {id}")
+                            hmms.write(line)
+
+
+class _EggnogHTMLParser(HTMLParser):
+    def __init__(self):
+        super().__init__()
+        self.links = []
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'a':
+            for attr in attrs:
+                if attr[0] == 'href':
+                    match = re.search(r'(\d+)/', attr[1])
+                    if match:
+                        self.links.append(match.group(1))
```

Comment:
```suggestion
                    if re.search(r'(\d+)/', attr[1]):
                        self.links.append(match.group(1))
```

Reply:
@misialq in this suggestion `match` is referenced before assignment.

Reply:
Oh, of course, I see now - never mind 🤦